[jacob.hyeon] 방탈출 인증관리 1,2 단계 리뷰 요청드립니다!

kjaehyeon opened the pull request:
[기능요구사항]
- 예약하기, 예약취소 개선 (아래의 API 설계에 맞춰 API 스펙을 변경한다.)
  - 비로그인 사용자는 예약이 불가능하다.
  - 자신의 예약이 아닌 경우 예약 취소가 불가능하다.
- 토큰 발급하는 API 생성
- 내 정보 조회하기 (토큰을 이용하여 본인 정보 응답하기)

[프로그래밍 요구사항]
- HandlerMethodArgumentResolver를 활용한다.

-> 테스트 및 구현 코드들이 사전에 모두 주어졌고, 제가 구현한 부분은 auth, config, exception 패키지 내부 클래스들과 ReservationService의 예약 삭제 메서드여서 이 부분만 리뷰해주시면 될 것 같습니다!

kjaehyeon committed:
feat: AuthController 구현 & 롬복 의존성 추가

kjaehyeon committed:
feat: AuthService 토큰 생성 메서드 구현

kjaehyeon committed:
feat: MemberController 구현

kjaehyeon committed:
feat: MemberService 구현

kjaehyeon committed:
test: Member 조회 테스트 작성

kjaehyeon committed:
feat: LoginInterceptor & LoginArgumentResolver 작성

kjaehyeon committed:
feat: 커스텀 예외 정의

kjaehyeon committed:
test: AuthE2ETest 요구사항에 맞게 수정

kjaehyeon committed:
feat: 멤버 로그인시 예외처리 구현

kjaehyeon committed:
test: 테스트 api 요청시 토큰 헤더에 추가하도록 수정

kjaehyeon committed:
test: Member 조회 테스트 작성

kjaehyeon committed:
feat: 자신이 예약이 아닌 경우 예약 취소가 불가능한 테스트 추가 및 구현

kjaehyeon committed:
refactor: Exception 단순화 & 메서드 분리

kjaehyeon committed:
refactor: jwt secretkey 코드와 분리

devk0ng commented on src/main/java/nextstep/config/LoginArgumentResolver.java:
```diff
@@ -35,7 +36,7 @@ public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer m
 
         if (request.getHeader(TOKEN_KEY).isBlank() || ObjectUtils.isEmpty(member)) {
-            throw new AuthorizationException();
+            throw new RoomEscapeException(ErrorCode.NOT_AUTHENTICATED);
```
제이콥~! 여기서 UNAUTHORIZED(401)을 뱉는거에 대해 조금 의문이 들어 코멘트 남깁니다.
지금 코드만 봤을 때에는 interceptor에서 authorization 헤더값에 대한 유무만 판단하는 것으로 보이네요.
토큰을 validate하는 코드도 작성하셨는데 사용하지 않은걸로 봐서 이 부분을 interceptor나 혹은 그에 준하는 곳에서 validation을 할 계획을 가지고 계신 것으로 이해했습니다!
argument resolver에서 만약 member가 없다면은 아마 인증단계는 지난 이후라 생각이 들어 개인적으로는 NOT_FOUND(404)가 어울리지 않을까 생각하는데 어떻게 생각하실까욥??

kjaehyeon replied:
앗 validation을 놓쳤네요.. LoginInterceptor에 토큰 validation 추가하도록 하겠습니다!

devk0ng commented on src/main/java/nextstep/config/LoginInterceptor.java:
```diff
@@ -0,0 +1,19 @@
+            throw new AuthorizationException();
+        }
+
+        return HandlerInterceptor.super.preHandle(request, response, handler);
```
이 부분과 더불어 WebMvcConfiguration 파일 내에
```java
WebMvcConfigurer.super.addInterceptors(registry);
...
WebMvcConfigurer.super.addArgumentResolvers(resolvers);
```
위와 같은 코드가 남아 있는데 불필요한 것으로 보이는데 혹시 용도가 있어서 남겨두신걸까요?

kjaehyeon replied:
IDE에서 자동구현 기능 사용했을때 자동으로 추가된 코드를 안지운것 같습니다.. 필요 없는 코드라고 생각되어 삭제하도록하겠습니다!

devk0ng replied:
코멘트 대상 코드인
```java
HandlerInterceptor.super.preHandle(request, response, handler)
```
요 녀석두 필요 없는 것 같습니다! (return true로 변경하면 되지 않을까 생각합니닷)

devk0ng reviewed (changes requested):
수고하셨습니다 제이콥~!
점점 시간이 갈수록 리뷰할 내용이 줄어들고 있네요ㅎㅎ 👍 
간단한 코멘트 두가지 남겼습니다. 확인부탁드려요~

kjaehyeon committed:
refactor: WebMvcConfiguration 에서 불필요한 코드 제거

kjaehyeon committed:
refactor:
- LoginInterceptor 에 토큰이 유효하지 않을 경우 예외 반환처리
- argumentResolver에서 멤버가 없을 경우 httpstatus 404 반환하도록 변경

kjaehyeon committed:
refactor: Interceptor 에 불필요한 코드 제거

devk0ng commented:
수고하셨습니다~! 머지할게요!
